### PR TITLE
Batched mult transpose in LOR transfer

### PR DESCRIPTION
**Pull request by pazner**

### CODE REVIEW

File: linalg/densemat.hpp [hunk 1]
```diff
@@ -1174,6 +1174,31 @@ class DenseTensor
...
+   /// dimensions @a i, @a j, and @a k.
+   ///
+   /// If @a own_mem is false, the DenseTensor will not own any of the pointers
+   /// of @a mem.
+   ///
+   /// Note that when @a own_mem is true, the @a mem object can be destroyed
+   /// immediately by the caller but `mem.Delete()` should NOT be called since
+   /// the DenseTensor object takes ownership of all pointers owned by @a mem.
+   void NewMemoryAndSize(const Memory<real_t> &mem, int i, int j, int k,
+                         bool own_mem)
```
Comment (tomstitt) [inline linalg/densemat.hpp:1187]: this is great!

**Comment by artv3** [comment]
Awesome super fast!

**Review by artv3 — APPROVED** [review]

**Comment by v-dobrev** [comment]
I tested a CUDA build of this branch and I saw no issues.

File: fem/transfer.cpp [hunk 2]
```diff
@@ -1138,37 +1025,15 @@ void L2ProjectionGridTransfer::L2ProjectionL2Space::ProlongateTranspose(
...
-                  dot += v_P(iho, ilo, iref, e) * v_x(iho, v, e);
-               }
-               v_y(ilo, iref, v, e) = dot;
-            }
-         }
-      }
-   });
+   DenseTensor P_dt;
+   P_dt.NewMemoryAndSize(P.GetMemory(), ndof_ho, ndof_lor * nref, nel_ho, false);
+   BatchedLinAlg::MultTranspose(P_dt, x, y);
```
Comment (artv3) [inline fem/transfer.cpp:1036]: I think we are missing a loop over vdim right?

Comment (pazner) [reply fem/transfer.cpp:1036]: It should be automatic based on the sizes (that's the idea anyway)

**Pull request merged**